sklearn/model_selection/tests/test_split.py

```diff
@@ -880,10 +880,15 @@ def test_cv_iterable_wrapper():
     # results
     kf_randomized_iter = KFold(n_splits=5, shuffle=True).split(X, y)
     kf_randomized_iter_wrapped = check_cv(kf_randomized_iter)
-    assert_array_equal(list(kf_randomized_iter_wrapped.split(X, y)),
-                       list(kf_randomized_iter_wrapped.split(X, y)))
-    assert_true(np.any(np.array(list(kf_iter_wrapped.split(X, y))) !=
-                       np.array(list(kf_randomized_iter_wrapped.split(X, y)))))
+    np.testing.assert_array_equal(
+        list(kf_randomized_iter_wrapped.split(X, y)),
+        list(kf_randomized_iter_wrapped.split(X, y)))
+    try:
+        np.testing.assert_equal(
+            np.array(list(kf_iter_wrapped.split(X, y))),
+            np.array(list(kf_randomized_iter_wrapped.split(X, y))))
+    except AssertionError:
```

Comment:
Ah sorry you asked this before and I failed to give a response!

So the nested lists comparison raises a deprecation warning...\

```py
>>> assert_true(np.array(list(kfold.split(X, y))) != np.array(list(kfold.split(X, y))))
DeprecationWarning: elementwise != comparison failed; this will raise an error in the future.
```

`np.testing.assert_equal` on the other hand handles list of np.ndarrays gracefully...